### DISCUSSION TITLE
Priority list

### DISCUSSION THREAD

**Discussion by EDsteve** [LIFsCode/ELOC-3.0#59] [General]
@LIFsCode @OOHehir 

The aim is to have a device that can record and detect 24/7. 
The only priority i can see now is to find and fix bugs which may interrupt this and as usual always have power consumption in the back of your mind :)

I will have a few ELOCs running with the latest commit for the next days.

**Comment by OOHehir** [comment]
@LIFsCode @EDsteve Over the last number of days I've pushed some commits to move dynamically allocated objects to statically allocated. It should help with an overview of memory usage, as suggested by @LIFsCode.

It seems at this point the code is relatively stable. Should we tag a release so we can continue to move forward with a known good reference point? I think there's plenty of clean-up that can be done that won't improve stability but should make code maintenance easier in the long term.

**Reply by LIFsCode** [reply]
@OOHehir  yes that's a good point.

If @EDsteve  does not have any major issues we can tag this as stable version and start cleanup from there on.

**Reply by EDsteve** [reply]
@LIFsCode @OOHehir 
Sorry for the late reply. Was on the road for a week and just arrived back home today.
It's definitely more stable than a few weeks ago. But i wouldn't call it really stable yet :)
Have some friends here tomorrow. But i will give my report about my expeience with the long term recordings/detecting as soon as i find time. I did bring only three devices with me on my trip though which i used for testing. 
Different commits give me different results and to make long term tests with every commit on a few devices is difficult, because too many commits :)

For example one commit can record/detect forever without problems and sometimes it doesn't. So it's gonna be difficult.

Most of my tests the last days i did with this commit: 0fbf78037fb15d81b67e9988404505de88315c81
And in short: It runs stable. But when BT gets involved (actiavated from the knock sensor) it sometimes reboots.

The new commit (d2dd26c8f8e37bfb0c3e9861ef266469df3d2101) i have only tested briefly yet and get different results. 

I basically need more time to test every commit. Can you give me advice which commits i should conentrate on?

**Comment by OOHehir** [comment]
I believe you should use the latest commit. I'll try to reduce the commits to simple bug fixes. 

I've started a separate [discussion](#70) on the issue of testing to keep the discussions ordered.

**Comment by OOHehir** [comment]
@LIFsCode @EDsteve In the absence of somewhere better to have this discussion I'll put this here rather than continuing on a closed issue (#76)

I've just made a commit to master, it's to check the SD card before opening a wav file, I was hoping to use to #74 to implement this but looks like it'll have to wait until after the imminent deployment. @EDsteve if you could please use 0079939eda11a68aff8a879ed7177431f17edd89 for testing if it's not too late.

**Reply by EDsteve** [reply]
@OOHehir 
Thanks. Yes. I will have one more day for testing. Tomorrow the ELOCs will be on their way to the Jungle.

**Comment by LIFsCode** [comment]
@OOHehir @EDsteve after the deployment of the ELOCs this week I would like to coordinate the work a bit.

@OOHehir AI would like to clean up the tasks in the system:
- I would prefer static tasks for the essential system tasks (recording, main, I2S, AI) keeping heap more static and more predictable memory usage
- clean up some inter task communication
- maybe integrate what's now in BT Server task into main, this does not actually require a separate task as recording and I2S is now separate from main

@OOHehir if you want we can make a short Skype/signal/whatever call to discuss on the architecture or just handle it via GitHub whatever you prefer

**Comment by OOHehir** [comment]
@LIFsCode Having a discussion regarding rearranging is a good idea. @EDsteve, could you please pass along my email address?
